include/grpc/support/thd.h

```diff
@@ -66,9 +65,25 @@ int gpr_thd_new(gpr_thd_id *t, void (*thd_body)(void *arg), void *arg,
 /* Return a gpr_thd_options struct with all fields set to defaults. */
 gpr_thd_options gpr_thd_options_default(void);
 
+/* Set the thread to become detached on startup - this is the default. */
+void gpr_thd_options_set_detached(gpr_thd_options *options);
```

Comment:
Any reason why detached should be the default? I'd personally find joinable more sensible for the default since that is the default for std::thread, pthreads, etc.